example/main.ts

```diff
@@ -4,79 +4,105 @@ import { Viewer } from './viewer';
-let loaded: boolean = false;
-
-const unloadBtn = document.createElement('button');
-unloadBtn.textContent = 'Unload';
-unloadBtn.addEventListener('click', () => {
-  if (!loaded) {
-    return;
-  }
-
-  viewer.unload();
-  loaded = false;
-  pointCloud = undefined;
-});
-
-const loadBtn = document.createElement('button');
-loadBtn.textContent = 'Load';
-loadBtn.addEventListener('click', () => {
-  if (loaded) {
-    return;
-  }
-
-  loaded = true;
-
-  viewer
-    .load(
-      'cloud.js',
-      'https://raw.githubusercontent.com/potree/potree/develop/pointclouds/lion_takanawa/',
-    )
-    .then(pco => {
-      pointCloud = pco;
-      pointCloud.rotateX(-Math.PI / 2);
-      pointCloud.material.size = 1.0;
-
-      pointCloud.material.clipMode = ClipMode.CLIP_HORIZONTALLY;
-      pointCloud.material.clipExtent = [0.0, 0.0, 0.5, 1.0];
-
-      const camera = viewer.camera;
-      camera.far = 1000;
-      camera.updateProjectionMatrix();
-      camera.position.set(0, 0, 10);
-      camera.lookAt(new Vector3());
-
-      viewer.add(pco);
-    })
-    .catch(err => console.error(err));
-});
-
-const slider = document.createElement('input');
-slider.type = 'range';
-slider.min = String(10_000);
-slider.max = String(500_000);
-slider.className = 'budget-slider';
-
-slider.addEventListener('change', () => {
-  if (!pointCloud) {
-    return;
-  }
-
-  pointCloud.potree.pointBudget = parseInt(slider.value, 10);
-  console.log(pointCloud.potree.pointBudget);
-});
-
-const btnContainer = document.createElement('div');
-btnContainer.className = 'btn-container';
-document.body.appendChild(btnContainer);
-btnContainer.appendChild(unloadBtn);
-btnContainer.appendChild(loadBtn);
-btnContainer.appendChild(slider);
+interface PointClouds {
+    [key: string]: PointCloudOctree | undefined;
+}
+
+interface LoadedState {
+    [key: string]: boolean;
+}
+
+const pointClouds: PointClouds = {
+    v1: undefined,
+    v2: undefined
+};
+
+const loaded: LoadedState = {
+    v1: false,
+    v2: false
+};
+
+function createButton(text: string, onClick: () => void): HTMLButtonElement {
+    const button: HTMLButtonElement = document.createElement('button');
+    button.textContent = text;
+    button.addEventListener('click', onClick);
+    return button;
+}
+
+function createSlider(version: string): HTMLInputElement {
+    const slider: HTMLInputElement = document.createElement('input');
+    slider.type = 'range';
+    slider.min = '10000';
+    slider.max = '500000';
+    slider.className = 'budget-slider';
+    slider.addEventListener('change', () => {
+        const cloud = pointClouds[version];
+        if (!cloud) {
+            return;
+        }
+        cloud.potree.pointBudget = parseInt(slider.value, 10);
+        console.log(cloud.potree.pointBudget);
+    });
+    return slider;
+}
+
+function setupPointCloud(version: 'v1' | 'v2', file: string, url: string): void {
+    if (loaded[version]) {
+        return;
+    }
+    loaded[version] = true;
```

Comment:
For an example, it would probably feel more natural to check if the url is loaded rater than the version

Reply:
Thanks for the suggestions @turien-pix4d  I've updated the unloading accordingly. In terms of checking the url, I personally think we can leave it as is as this is really just meant as an example to show how to load and dump the two different Pointcloud versions.